src/responder/kcm/kcm_renew.c

```diff
@@ -0,0 +1,297 @@
+extern struct dp_option default_krb5_opts[];
+
+static int kcm_get_auth_provider_options(struct kcm_ctx *kctx,
+                                         struct krb5_ctx *krb5_ctx)
+{
+    errno_t ret = 0;
+    char *lifetime_str = NULL;
+    char *rtime = NULL;
+    bool validate = false;
+    bool canonicalize = false;
+    int child_timeout = 0;
+    struct dp_option *opts;
+    const char *conf_path = NULL;
+    char *auth_provider = NULL;
+    struct sss_domain_info *domains = NULL;
+    struct sss_domain_info *dom = NULL;
+
+   ret = confdb_get_domains(kctx->rctx->cdb, &domains);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Cannot get domains\n");
+        goto done;
+    }
+
+    for (dom = domains; dom != NULL;
+            dom = get_next_domain(dom, SSS_GND_DESCEND)) {
+
+        conf_path = talloc_asprintf(kctx->rctx, CONFDB_DOMAIN_PATH_TMPL,
+                                    dom->name);
+        if (conf_path == NULL) {
+            DEBUG(SSSDBG_CRIT_FAILURE, "Out of memory\n");
+            ret = ENOMEM;
+            goto done;
+        }
+
+        ret = confdb_get_string(kctx->rctx->cdb,
+                                kctx->rctx,
+                                conf_path,
+                                CONFDB_DOMAIN_AUTH_PROVIDER,
+                                0, &auth_provider);
+
+        if (auth_provider != NULL) {
+            if (strcasecmp(auth_provider, "krb5") == 0) {
```

Comment:
This needs to be at least documented.

I'm unsure about this functionality, it will cause troubles in multi-domain setup. Ideally we should match the ccache to a domain, then read configuration from the domain. But since this would be too difficult, imho it is best to allow configuration only in [kcm] instead of trying to be smart. (I did not go through all the comments here, I apologize if it was already discussed somewhere).

Reply:
Okay thanks for the input, it was a recommendation by @sumit-bose - Sumit do you have strong opposition to removing the fallback to `auth_provider = krb5` options for KCM renewals?

Reply:
I updated the man page to ensure that the fallback any existing auth_provider=krb5 is documented.